build.gradle

```diff
@@ -70,15 +70,11 @@ allprojects {
         toolchain {
             languageVersion = JavaLanguageVersion.of(21)
         }
+        sourceCompatibility = 17    // javac --source
+        targetCompatibility = 17    // javac --target
     }
-
-    compileJava {
-        options.release.set(17)
-    }
-
-    tasks.withType(GroovyCompile) {
-        sourceCompatibility = '17'
-        targetCompatibility = '17'
+    tasks.withType(JavaCompile).configureEach {
+        options.release = 17        // javac --release
     }
```

Comment:
How this syntax improves? afaik `withType(JavaCompile)` does not apply to Java code compiled via groovy compiler